testsuites/syncgateway/functional/tests/test_log_rotation_new.py

```diff
@@ -0,0 +1,609 @@
+
+    SG_LOGGING_SECTIONS = ['console', 'error', 'info', 'warn', 'debug']
+
+    # Remove the rotation key from sample config
+    for section in SG_LOGGING_SECTIONS:
+        del data['logging'][section]["rotation"]
+
+    # Create temp config file in the same folder as sg_conf
+    temp_conf = "/".join(sg_conf.split('/')[:-2]) + '/temp_conf.json'
+
+    log_info("TEMP_CONF: {}".format(temp_conf))
+
+    with open(temp_conf, 'w') as fp:
+        json.dump(data, fp, indent=4)
+
+    # Create /tmp/sg_logs
+    sg_helper.create_directory(cluster_config=cluster_conf, url=sg_one_url, dir_name="/tmp/sg_logs")
+
+    SG_LOGS = ['sg_debug', 'sg_error', 'sg_info', 'sg_warn']
+    remote_executor = RemoteExecutor(cluster.sync_gateways[0].ip)
+
+    for log in SG_LOGS:
+        # Generate a log file with size ~94MB to check that backup file not created while 100MB not reached
+        file_size = 94 * 1024 * 1024
+        file_name = "/tmp/sg_logs/{}.log".format(log)
+        log_info("Testing log rotation for {}".format(file_name))
+        sg_helper.create_empty_file(cluster_config=cluster_conf, url=sg_one_url, file_name=file_name, file_size=file_size)
+
+    # iterate 5th times to verify that every time we get new backup file with ~100MB
+    for i in xrange(5):
+        sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=temp_conf)
+        # ~1M MB will be added to the info/debug/warn log files after the requests
+        remote_executor.execute(
+            "for ((i=1;i <= 2000;i += 1)); do curl -s http://localhost:4984/ABCD/ > /dev/null; done")
+        remote_executor.execute(
+            "for ((i=1;i <= 2000;i += 1)); do curl -s -H 'Accept: text/plain' http://localhost:4985/db/ > /dev/null; done")
+
+        # Verify num of log files for every log file type
+        for log in SG_LOGS:
+            file_name = "/tmp/sg_logs/{}.log".format(log)
+            _, stdout, _ = remote_executor.execute("ls /tmp/sg_logs/ | grep {} | wc -l".format(log))
+            log_info("Checking for {} files for {}".format(i + 1, file_name))
+            assert stdout[0].rstrip() == str(i + 1)
+
+            sg_helper.stop_sync_gateways(cluster_config=cluster_conf, url=sg_one_url)
+
+            # Generate an empty log file with size ~100MB
+            file_size = 100 * 1024 * 1024
+            sg_helper.create_empty_file(cluster_config=cluster_conf, url=sg_one_url, file_name=file_name, file_size=file_size)
+
+    sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=sg_conf)
+
+    # Remove generated conf file
+    os.remove(temp_conf)
+
+
+@pytest.mark.syncgateway
+@pytest.mark.logging
+@pytest.mark.parametrize("sg_conf_name", ["log_rotation_new"])
+def test_invalid_logKeys_string(params_from_base_test_setup, sg_conf_name):
+    """Negative test to verify that we are not able start SG when
+    logKeys is string
```

Comment:
is it possible to add more tests with different combination of log keys, not just strings?

Reply:
This is a negative test, log key as a string should fail. We are not testing log keys here.